### PR TITLE
rgw: drop unused vector elements

### PR DESCRIPTION
**Pull request by qrGitHub**

### CODE REVIEW

**Comment by mattbenjamin** [comment]
@qrGitHub I don't know this code path to speak of, but can you provide some quantitative reasoning for why not using the larger size hint isn't usually an optimization?  often, it would be, one would think

**Comment by qrGitHub** [comment]
@mattbenjamin The former list_results.size() elements in vcurrents/vends/vnames are not initialized, so they are useless here.

**Comment by qrGitHub** [comment]
Hi @yehudasa, would you help to review? Thanks.

**Comment by qrGitHub** [comment]
Hi @yehudasa @cbodley, would you help to review this? Thanks.

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -12831,9 +12831,9 @@ int RGWRados::cls_bucket_list(RGWBucketInfo& bucket_info, int shard_id, rgw_obj_
     return r;
 
   // Create a list of iterators that are used to iterate each shard
-  vector<map<string, struct rgw_bucket_dir_entry>::iterator> vcurrents(list_results.size());
-  vector<map<string, struct rgw_bucket_dir_entry>::iterator> vends(list_results.size());
-  vector<string> vnames(list_results.size());
+  vector<map<string, struct rgw_bucket_dir_entry>::iterator> vcurrents;
+  vector<map<string, struct rgw_bucket_dir_entry>::iterator> vends;
+  vector<string> vnames;
```
Comment (cbodley) [inline src/rgw/rgw_rados.cc]: it still makes sense to prealloacate the storage for this vector with `vector::.reserve()` so it doesn't have to reallocate as it grows

Comment (qrGitHub) [reply src/rgw/rgw_rados.cc]: Yes, you are right @cbodley.
"reserve" is added, would you help to review again? Thanks.

**Comment by qrGitHub** [comment]
Hi @cbodley, can this be merged? Thanks.

**Comment by qrGitHub** [comment]
Hi @cbodley, can this be merged? Thanks.

**Comment by cbodley** [comment]
@qrGitHub i'd like to have @yuriw run this through testing first

**Comment by qrGitHub** [comment]
Hi @cbodley, can this be merged? Thanks.

**Comment by yuriw** [comment]
[wip-yuri3-testing-2018-02-14-2124](https://trello.com/c/guwy7LoB/261-wip-yuri3-testing-2018-02-14-2124)

**Pull request merged**